lib/asciidoctor/inline_parser/mapper.rb

```diff
@@ -0,0 +1,94 @@
+require_relative 'inline_node'
+
+# rubocop:disable all
+module Asciidoctor
```

Comment:
Do not look at this code please :see_no_evil: 
The important part is the structure of the inline AST (the end result) and what the AST looks like for a given input.